odoo/fields.py

```diff
@@ -2583,6 +2583,44 @@ def convert_to_display_name(self, value, record):
         return ustr(value and value.display_name)
 
 
+class Domain(Field):
+    """
+    Field that contain a domain on a given model
+
+    :param str domain_model: name of the model on which the domain apply.
+    :param str domain_model_field: name of the :Class:`Char`, :class:Selection, or :class:`Many2one` where the model name is stored.
+    """
+
+    type = 'domain'
+    column_type = ('varchar', pg_varchar())
+
+    default = "[]"
+    required = True
```

Comment:
Why?  No other field has a forced default, no other field is forcibly required.
It will cause programmer confusion.  Explicit is better in this case.
Several of your examples actually change the database schema, and it is not visible in the diff!
(BTW it is also the reason why the build fails.)

Reply:
You're right. Making this field required by default was an error.